### PR TITLE
Add e2e for telemetry filters with wasm runtime

### PR DESCRIPTION
**Pull request by bianpengyuan**

### CODE REVIEW

File: tests/integration/telemetry/stats/prometheus/http/wasm/stats_filter_wasm_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+func setupConfig(cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	// disable mixer telemetry and enable telemetry v2 with Wasm
+	cfg.Values["telemetry.enabled"] = "true"
+	cfg.Values["telemetry.v1.enabled"] = "false"
+	cfg.Values["telemetry.v2.enabled"] = "true"
+	cfg.Values["telemetry.v2.metadatExchange.wasmEnabled"] = "true"
```
Comment (howardjohn) [inline tests/integration/telemetry/stats/prometheus/http/wasm/stats_filter_wasm_test.go]: typo on metadata

Comment (bianpengyuan) [reply tests/integration/telemetry/stats/prometheus/http/wasm/stats_filter_wasm_test.go]: Done, thanks!

**Comment by bianpengyuan** [comment]
Hmm looks like telemetry test is stuck on a KinD failure.

File: manifests/istio-control/istio-discovery/values.yaml [hunk 2]
```diff
@@ -174,9 +174,12 @@ telemetry:
     # For Null VM case now. If enabled, will set disableMixerHttpReports to true and not define mixerReportServer
     # also enable metadata exchange and stats filter.
     enabled: true
+    metadataExchange:
```
Comment (mandarjog) [inline manifests/istio-control/istio-discovery/values.yaml]: For clarity fix the comments above.

Comment (mandarjog) [reply manifests/istio-control/istio-discovery/values.yaml]: Bike shed: “useWasm” ?

Comment (bianpengyuan) [reply manifests/istio-control/istio-discovery/values.yaml]: This PR is to track e2e test of the Wasm filter and the option will be added in https://github.com/istio/istio/pull/21776. I will address your comment there.

Regarding to the name.. both names make sense to me so I don't have much preference :) `wasmEnabled` is more consistent with other name in the telemetry block, so I'll stick with it.

**Comment by bianpengyuan** [comment]
Surprisingly, nullvm based test fails due to bookinfo deployment times out, which looks more like a wasm runtime issue. From the log, the configuration for the nullvm test looks correct though...

**Comment by bianpengyuan** [comment]
@istio/wg-policies-and-telemetry-maintainers ptal, thanks!

**Review by kyessenov — APPROVED** [review]

**Pull request merged**